### PR TITLE
XW-3650 | Featured Video on scroll unit should have priority over other page elements

### PR DESCRIPTION
**Pull request by JamesHawking**
[Ticket](https://wikia-inc.atlassian.net/browse/XW-3650)

We decided to hide site wide messages on Featured Video pages so they do not interfere with video experience.

@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/SiteWideMessages/SiteWideMessagesController.class.php [hunk 1]
```diff
@@ -1,14 +1,30 @@
 <?php
 
-class SiteWideMessagesController extends WikiaController  {
+class SiteWideMessagesController extends WikiaController {
 
 	const CACHE_VALIDITY_VARNISH = 10800; // 3 hours
 	const CACHE_VALIDITY_BROWSER = 3600; // 1 hour
 
+	private function shouldShowSiteWideMessage() {
```
Comment (hakubo) [inline extensions/wikia/SiteWideMessages/SiteWideMessagesController.class.php]: shouldSkip

File: extensions/wikia/SiteWideMessages/SiteWideMessagesController.class.php [hunk 2]
```diff
@@ -1,14 +1,30 @@
...
+			$isFeaturedVideoEmbedded = false;
+		}
+
+		return $this->wg->User->isLoggedIn() || $isFeaturedVideoEmbedded;
+	}
+
 	public function getAnonMessages() {
-		if ( $this->wg->User->isLoggedIn() ) {
+
+		if ( $this->shouldShowSiteWideMessage() ) {
```
Comment (rybmat) [inline extensions/wikia/SiteWideMessages/SiteWideMessagesController.class.php]: the method name says something opposite than it does ;) please change it to shouldNotShowSiteWideMessages()

File: extensions/wikia/SiteWideMessages/SiteWideMessagesController.class.php [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
 
 	const CACHE_VALIDITY_VARNISH = 10800; // 3 hours
 	const CACHE_VALIDITY_BROWSER = 3600; // 1 hour
 
+	private function shouldShowSiteWideMessage() {
+		global $wgEnableArticleFeaturedVideo;
+
+		$title = Title::newFromID( $this->getVal( 'articleId' ) );
+		if ( !is_null( $title ) ) {
+			$isFeaturedVideoEmbedded = !empty( $wgEnableArticleFeaturedVideo ) &&
```
Comment (hakubo) [inline extensions/wikia/SiteWideMessages/SiteWideMessagesController.class.php:13]: actually there is no need to call newFromId is that is false

**Review by hakubo — CHANGES_REQUESTED** [review]

**Review by rybmat — CHANGES_REQUESTED** [review]

File: extensions/wikia/SiteWideMessages/SiteWideMessagesController.class.php (hunk 3 above)
Comment (JamesHawking) [reply extensions/wikia/SiteWideMessages/SiteWideMessagesController.class.php:13]: but we need title to check it if i understand correctly

**Review by hakubo — APPROVED** [review]

**Review by rybmat — APPROVED** [review]

**Pull request merged**